### PR TITLE
HelloEmbeddedEngineering: Added basic content

### PR DESCRIPTION
**Pull request by pavly-gerges**
## Contents: 

- [x] Volatge, Current and basic components.
- [x] Embedded engineering principles questions.
- [x] Embedded basic C/C++.
<s>

- [ ] GNU C Compiler (GCC) and Scripting languages questions.
- [ ] Advanced embedded engineering questions.
- [ ] Advanced C/C++.
- [ ] Design patterns and algorithms.
<s/>

### CODE REVIEW

**Comment by Scrappers-glitch** [comment]
We are going to defer those tasks: 
- GNU C Compiler (GCC) and Scripting languages questions.
- Advanced embedded engineering questions.
- Advanced C/C++.
- Design patterns and algorithms.

To another PR.....

File: HelloEmbeddedEngineering/README.md [hunk 1]
```diff
@@ -0,0 +1,481 @@
+# Contents: 
+- Volatge, Current and basic components (circuits, resistors, transistors, batteries).
+- Embedded engineering principles questions (boolean algebra and digital electronics).
+- Embedded basic C/C++ (data types, variables, pointers, buffers and functions)
+- GNU C Compiler (GCC) and Scripting languages questions.
+- Advanced embedded engineering questions.
+- Advanced C/C++.
+- Design patterns and algorithms.
+
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:9]: Done.

File: HelloEmbeddedEngineering/README.md [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+---------------------------------------------------------
+
+## Volatge, Current and basic components:
+
+1) Resistors are used for which of the following:
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:14]: @HassanAhmed093
Please add this one.

File: HelloEmbeddedEngineering/README.md [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+
+1) Resistors are used for which of the following: 
+    - [ ] Current limiting devices according to Ohm's law.
+    - [ ] Current limiting devices according to KCL.
+    - [ ] Voltage divider devices according to KVL.
+    - [ ] Current dividers accoridng to KCL.
+    - [x] A, C & D.
+
+2) The following circuit shows the usage of resistors as -----------
+
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:22]: @HassanAhmed093
And this one.

File: HelloEmbeddedEngineering/README.md [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+    ![image](https://user-images.githubusercontent.com/60224159/185788864-d9df48d9-a705-464d-8474-10d0a3d011cf.png)
+
+    - [ ] Current limiting devices according to Kirchhoff's current law (KCL).
+    - [ ] Voltage limiting devices.
+    - [x] Current limiting devices according to ohm's law.
+    - [ ] A and C.
+
+4) In the following circuit, V(P) can be solved using -------------
+
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:40]: @HassanAhmed093
And this too !

File: HelloEmbeddedEngineering/README.md [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+8) The following circuit shows the operation of batteries in series ? 
+
+    ![image](https://user-images.githubusercontent.com/60224159/185791637-082f7aae-cafc-4b81-813d-3d0c3265225d.png)
+
+    - [x] true.
+    - [ ] false.
+
+9) The following circuit aims at increasing the circuit voltage and current ?
+
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:79]: @HassanAhmed093

This one too.

File: HelloEmbeddedEngineering/README.md [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+    - [ ] false.
+
+10) To control the speed of a motor manually, one should use a/an ------------
+    - [x] Electric Potentiometer.
+    - [ ] Insulator.
+    - [ ] Resistor.
+    - [ ] Capacitor.
+
+11) The following circuit, shows the operation of transistors as a switch with a current gain of hfE or Beta ? 
+
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:92]: @HassanAhmed093
And this one, as well.

File: HelloEmbeddedEngineering/README.md [hunk 7]
```diff
@@ -0,0 +1,481 @@
...
+	 where R(2) is the total resistance value for the CE circuit and hFE is the current gain for the 
+         transistor. 
+```
+
+--------------------------------------
+
+## Embedded engineering principles questions:
+
+1) What's the usage of the AND gate ? 
+
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:137]: @HassanAhmed093
Please, add this.

File: HelloEmbeddedEngineering/README.md [hunk 8]
```diff
@@ -0,0 +1,481 @@
...
+	- [ ] Adding 2 bytes together.
+	- [ ] Subtracting 2 bytes from one another.
+	- [x] Multiplying 2 bytes.
+	- [ ] A and C.
+
+2) What's the usage of the OR gate ? 
+
+	- [ ] Adding 2 bytes togther.
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:145]: @HassanAhmed093
This question, as well.

File: HelloEmbeddedEngineering/README.md [hunk 9]
```diff
@@ -0,0 +1,481 @@
...
+	- [ ] OR-Gates.
+	- [ ] NAND-Gates.
+
+------------------------------------------------------------------
+
+## Embedded basic C/C++: 
+
+1) What's the output of this code: 
+```c
+#include<stdint.h>
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:238]: @HassanAhmed093
This one.

File: HelloEmbeddedEngineering/README.md [hunk 10]
```diff
@@ -0,0 +1,481 @@
...
+}
+```
+-
+	- [ ] 0b00000000
+	- [x] 0b01100110
+	- [ ] 0b01100000
+	- [ ] 0b01000110
+	
+4) What's the output of this code ? 
+```c
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:312]: @HassanAhmed093
And this question too.

File: HelloEmbeddedEngineering/README.md [hunk 11]
```diff
@@ -0,0 +1,481 @@
...
+```
+-
+	- [x] 0b01100000
+	- [ ] 0b00110000
+	- [ ] 0b00010000
+	- [ ] 0b00111000
+
+
+2) What's the output of this code: 
+```c
```
Comment (Scrappers-glitch) [inline HelloEmbeddedEngineering/README.md:254]: @HassanAhmed093
And this question, as well.

**Review by pavly-gerges — COMMENTED** [review]
@HassanAhmed093

Those are the suggested questions to add, let me know if you have any changes or thoughts.

Thanks.
pavl_g.

**Comment by Scrappers-glitch** [comment]
Going to merge this within 24 hrs if there are no more substantial reviews.

**Pull request merged**